projects/packages/videopress/src/class-initializer.php

```diff
@@ -304,14 +307,26 @@ public static function render_videopress_video_block( $block_attributes, $conten
 			);
 		}
 
+		// Get premium content from block context
+		$premium_block_plan_id    = isset( $block->context['premium-content/planId'] ) ? intval( $block->context['premium-content/planId'] ) : 0;
+		$is_premium_content_child = isset( $block->context['isPremiumContentChild'] ) ? (bool) $block->context['isPremiumContentChild'] : false;
+		$maybe_premium_script     = '';
+		if ( $is_premium_content_child ) {
+			Access_Control::instance()->set_guid_subscription( $guid, $premium_block_plan_id );
+			$escaped_guid         = esc_js( $guid );
+			$script_content       = "if ( ! window.__guidsToPlanIds ) { window.__guidsToPlanIds = {}; }; window.__guidsToPlanIds['$escaped_guid'] = $premium_block_plan_id;";
```

Comment:
That can simplified as well

```
$script_content = "window.__guidsToPlanIds = { '$escaped_guid': $premium_block_plan_id, ...window.__guidsToPlanIds };"

```

See this screenshot (totally legit)

<img width="285" alt="Screenshot 2023-10-27 at 12 44 50" src="https://github.com/Automattic/jetpack/assets/790558/557a7881-841f-46af-8c99-40a163247593">

Reply:
I was trying to be conservative with es6 features on this, since I am not sure where this will run.